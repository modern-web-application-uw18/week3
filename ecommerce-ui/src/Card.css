.Card-img {
	width: 430px;
}

.Card-title {
	font-size: 19px;
	font-weight: bold;
}

.flex-container {
  display: flex;
  flex-wrap: wrap;
  border-style: solid;
  border-width: 1px;
	border-color: lightgray;
	margin: 10px;
}

.flex-container > div {
  width: 500px;
	height: 450px;
  font-size: 15px;
}

.grid-card-top {
  margin: auto;
}

.grid-card-bottom {
	position: relative;
	margin: 9px;
}
