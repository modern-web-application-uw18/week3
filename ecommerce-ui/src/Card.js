import React, { Component } from 'react';
import PropTypes from 'prop-types';
import StarRatings from 'react-star-ratings';
import './CardStyle.css';


class Card extends Component{  
     
    Superhost(props){
        const isSuperhost = this.props.card.host.isSuperhost;
         if(isSuperhost){
             return <p className='Superhost'>*Superhost</p>;
         } 
    }
    
    render(){
     
        return(
      <div className='Card-container'>   
           <div className='Card-img'>
                <img className = 'theImage' 
                        alt='staging'  
                        src={this.props.card.image}/>       
            </div>
            
            <div className='House-type-city-title'>
                <div className='Card-housetype'>
                    {this.props.card.houseType}
                        * 
                    {this.props.card.location.city}
                </div>
                    <div className='Card-title'>
                    {this.props.card.title}
                    </div> 
            </div>        
            <div className='Card-payment'>${this.props.card.payment.cost} per night * {this.props.card.payment.description} 
            </div> {/*end card payment  */}
            <div className='Card-rating'>
                 <StarRatings className='Stars' 
                    rating={this.props.card.rating.stars} 
                    starRatedColor='blue' 
                    numberOfStars={5}
                    starDimension='15px'
                    starSpacing="0.5px" 
                 /> 
                 
               <div className='Number-reviews'> 
               <div className='reviews'>{this.props.card.rating.reviews}</div>
               {this.Superhost(this.props.card.host.isSuperhost)}
               </div> 
            </div> {/*end card rating */} 
       </div>
        );
    }

}


Card.propTypes = {
    card: PropTypes.shape({  
        title: PropTypes.string,
        houseType: PropTypes.string,
        image: PropTypes.string,
        location: PropTypes.shape({
            city: PropTypes.string,
            country:PropTypes.string
        }),
        payment: PropTypes.shape({
            cost: PropTypes.number,
            description: PropTypes.string
        }),
        host: PropTypes.shape({
            name: PropTypes.string,
            isSuperhost: PropTypes.bool
        }),
        rating: PropTypes.shape({
            stars: PropTypes.number,
            reviews: PropTypes.number
        })
    })
};


export default Card;