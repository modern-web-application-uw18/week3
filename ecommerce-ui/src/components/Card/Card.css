.Card {
    height: 200px;
    width: 500px;
    min-width: 40%;
    display: inline-flex;
    margin: 5px;
    background-position: 50% 50% !important;
    background-size: cover!important;
    text-align: left !important;
    padding: 5px;
    position: relative;
    flex-wrap: wrap;
    border: solid 1px #cecece;

}


.Title {
    background-color: #ffffff;
    color: black;
    width: 50%;
    height: 50%;
    padding: 5px;
    font-family: 'Roboto', sans-serif;
    font-weight: 300;
    font-style: italic;
    font-size: 25px;
    display: inline-flex;    
    opacity: 0.90;
}


.Rating {
    background-color: #be2c00;
    color: rgb(231, 231, 231);
    width: 50%;
    height: 18%;
    padding: 5px;
    font-family: 'Roboto', sans-serif;
    font-weight: 300;
    font-style: italic;
    font-size: 15px;
    display: inline-flex;    
    opacity: 0.90;
}


.UpperRight {
    background-color: #ffffff; 
    text-align: right;
    width: 48%;
    height: 50%;
    padding: 5px;
    font-family: 'Roboto', sans-serif;
    font-weight: 700;
    /* font-style: italic; */
    font-size: 45px;
    display: inline-flex; 
    margin-left: 5px;   
    opacity: 0.750;
}

.AlignRight {
    text-align: center; 
    width: 100%;
    color: #007e15;
    opacity: 1;

}

.LowerRight {
    background-color: #1b1a1a;
    color: rgb(231, 231, 231);
    width: 48%;
    height: 50%;
    padding: 5px;
    font-family: 'Roboto', sans-serif;
    font-weight: 300;
    font-style: italic;
    font-size: 15px;
    display: inline-flex; 
    margin-left: 5px;   
    opacity: 0.90;
}

.nopadding {
    padding-left: 0 !important;
    padding-right: 0 !important;
 }


@media (max-width: 650px) {
    .Card {
        width: 100%;
        margin: 0px;
    }

    .Title .Rating .UpperRight .LowerLeft {
        font-size: 20px;
    }


}
 
