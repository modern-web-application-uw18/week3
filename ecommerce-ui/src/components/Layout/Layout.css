.Layout{
    /* width: 100%; */
    /* margin: auto; */
    /* height: 100%; */
    /* text-align: center; */
    /* background-color: rgb(245, 245, 245);
} */

/* @media (min-width: 650px) {
    .Layout {
        width: 80%;
        height: 100%;
    }
} */

}