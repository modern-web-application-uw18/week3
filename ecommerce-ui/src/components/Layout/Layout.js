import React from 'react';
import Cart from '../../components/Cart/Cart'
import classes from './Layout.css';

const layout = (props) => (
    <div className={classes.Layout}>
      
        <Cart />

        <main>
            {props.children}
        </main>

        <div style={{display:'flex',height:'100%',backgroundColor:'red'}}></div>
    </div>
);

export default layout;
