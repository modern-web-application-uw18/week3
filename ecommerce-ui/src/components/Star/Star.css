.AlignRight {
    text-align: right; 
    width: 100%;
    color: #ffffff;
    opacity: .5;

}

.Star {
    height: 20px;
    width: 20px;
    margin-bottom: 5px;
}

@media (max-width: 668px) {
    .Star {
        height: 12px;
        width: 12px;
        margin-bottom: 2px;
    }

}