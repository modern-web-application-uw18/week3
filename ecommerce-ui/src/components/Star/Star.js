import React from 'react';
import Aux from '../../hoc/Auxl'
import classes from './Star.css';
import MdStar from 'react-icons/lib/md/star';
import MdStarHalf from 'react-icons/lib/md/star-half';


let starMaker = (starCount) => {
    let stars = [];
    starCount % 1 !== 0 ? stars.push("half") : null;
    while(starCount >= 1) {
        stars.push("whole");
        starCount--;
    }

    stars.reverse();

    return (
        <Aux>
            {stars.map((item,idx)=>{
              return item === 'whole' ? <MdStar size={20} key={idx} /> : <MdStarHalf size={20} key={idx} />              
            })}
        </Aux>
    )

}

const star = (props) => (

    <Aux>
        <div className={classes.AlignRight}>
            {starMaker(props.starCount)}
        </div>

    </Aux>


)

export default star;