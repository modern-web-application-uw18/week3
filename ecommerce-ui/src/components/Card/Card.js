import React from 'react';
import classes from './Card.css';
import MdAddShoppingCart from 'react-icons/lib/md/add-shopping-cart';
import Star from '../Star/Star';




const layout = (props) => (


  
    <div className={classes.Card} style={{ backgroundImage: `url(${props.item.image})` }}>

        <div className={classes.Title}>
            {props.item.title}
        </div>

        <div className={classes.UpperRight}>

            <div className={classes.AlignRight}>
                $ {props.item.payment.cost}
            </div>

        </div>

        <div className={classes.Rating}>
            <div style={{ width: '200px' }}>
                {props.item.houseType}
            </div>
            &nbsp;&nbsp;
            <Star starCount={props.item.rating.stars} />
            <span style={{ fontSize: '13px' }}>
                ({props.item.rating.reviews})
             </span>
        </div>


        <div className={classes.LowerRight}>


             <div>
                {props.item.location.city}, {props.item.location.city}
            </div>

            <div style={{ textAlign: 'left'}}>
                <div className={classes.Name}>
                  {props.item.host.name}
                </div>
            </div>

            <div style={{textAlign: 'right' }}>
                    <button type="button" className="btn btn-default" style={{width:'45px',height:'35px',backgroundColor:'#5b5b5b',border:'0px'}}><MdAddShoppingCart size={20} style={{color:'#ffffff'}} /></button>
            </div> 


        </div>

    </div>







    //         <div className={classes.Card} style={{ backgroundImage: `url(${props.item.image})` }}>

    //         <div className={classes.Title}>
    //         {props.item.title}
    //     </div> 

    // <div className='container' style={{paddingLeft: 0, paddingRight: 0}}>

    // <div className='container' className={classes.nopadding}>
    //         <div className='row'>
    //             <div className="col">
    //                 <div className={classes.Title}>
    //                     {props.item.title}
    //                 </div>
    //             </div>

    //             <div className="col">
    //             <div className={classes.UpperRight}>
    //                     {props.item.houseType}
    //                 </div>
    //              </div>
    //         </div>

    //         <div className='row'>
    //             <div className="col">

    //                 <div className={classes.Rating}>
    //                     {props.item.houseType}
    //                 </div>


    //             </div>

    //             <div className="col">
    //             <div className={classes.LowerLeft}>
    //                     {props.item.houseType}
    //                 </div>
    //              </div>
    //         </div>
    //     </div>
    //     </div>
    //     </div>

)


export default layout;
