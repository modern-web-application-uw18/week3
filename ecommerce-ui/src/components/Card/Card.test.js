import React from 'react';
import { shallow, mount} from 'enzyme';
import Card from './Card';
import Layout from '../../containers/Layout/Layout';
import mockListings from '../../assets/airbnbs.json';

import classes from './Card.css';
import MdAddShoppingCart from 'react-icons/lib/md/add-shopping-cart';
import Star from '../Star/Star';

describe('<Card />', () => {

    it('Snapshot test', () => {

        const wrapper2 = mount(<Layout  />);
        let fn = wrapper2.instance().addListingToTheCartHandler;

        const wrapper = mount(<Card item={mockListings[0]} addListingToTheCartHandler={fn}  />);
               
        expect(wrapper.find('.btn')).toHaveLength(1);
        wrapper.find('.btn').simulate('click');

        expect(wrapper).toMatchSnapshot();
    })
})