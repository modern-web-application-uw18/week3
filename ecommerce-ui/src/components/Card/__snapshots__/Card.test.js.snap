// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> Snapshot test 1`] = `
<card
  addListingToTheCartHandler={[Function]}
  item={
    Object {
      "host": Object {
        "isSuperhost": true,
        "name": "Robert",
      },
      "houseType": "Apartment",
      "image": "https://images.unsplash.com/photo-1490806230066-f7e6f7bf5052?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=2ca889535ea01f912f94ac4ddf0034e0&auto=format&fit=crop&w=500&q=80",
      "location": Object {
        "city": "New York",
        "country": "USA",
      },
      "payment": Object {
        "cost": 105,
        "description": "Free cancellation",
      },
      "rating": Object {
        "reviews": 133,
        "stars": 3,
      },
      "title": "Centrally-located Manhattan studio",
    }
  }
>
  <div
    style={
      Object {
        "backgroundImage": "url(https://images.unsplash.com/photo-1490806230066-f7e6f7bf5052?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=2ca889535ea01f912f94ac4ddf0034e0&auto=format&fit=crop&w=500&q=80)",
      }
    }
  >
    <div>
      Centrally-located Manhattan studio
    </div>
    <div>
      <div>
        $ 
        105
      </div>
    </div>
    <div>
      <div
        style={
          Object {
            "width": "200px",
          }
        }
      >
        Apartment
      </div>
        
      <star
        starCount={3}
      >
        <div>
          <MdStar
            key="0"
          >
            <IconBase
              viewBox="0 0 40 40"
            >
              <svg
                fill="currentColor"
                height="1em"
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "color": undefined,
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 40 40"
                width="1em"
              >
                <g>
                  <path
                    d="m20 28.8l-10.3 6.2 2.7-11.7-9-7.9 11.9-1 4.7-11 4.7 11 11.9 1-9 7.9 2.7 11.7z"
                  />
                </g>
              </svg>
            </IconBase>
          </MdStar>
          <MdStar
            key="1"
          >
            <IconBase
              viewBox="0 0 40 40"
            >
              <svg
                fill="currentColor"
                height="1em"
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "color": undefined,
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 40 40"
                width="1em"
              >
                <g>
                  <path
                    d="m20 28.8l-10.3 6.2 2.7-11.7-9-7.9 11.9-1 4.7-11 4.7 11 11.9 1-9 7.9 2.7 11.7z"
                  />
                </g>
              </svg>
            </IconBase>
          </MdStar>
          <MdStar
            key="2"
          >
            <IconBase
              viewBox="0 0 40 40"
            >
              <svg
                fill="currentColor"
                height="1em"
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "color": undefined,
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 40 40"
                width="1em"
              >
                <g>
                  <path
                    d="m20 28.8l-10.3 6.2 2.7-11.7-9-7.9 11.9-1 4.7-11 4.7 11 11.9 1-9 7.9 2.7 11.7z"
                  />
                </g>
              </svg>
            </IconBase>
          </MdStar>
        </div>
      </star>
      <span>
        (
        133
        )
      </span>
    </div>
    <div>
      <div>
        New York
        , 
        USA
      </div>
      <div
        style={
          Object {
            "textAlign": "left",
          }
        }
      >
        <div>
          Robert
        </div>
      </div>
      <div
        style={
          Object {
            "textAlign": "right",
          }
        }
      >
        <button
          className="btn btn-default"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#5b5b5b",
              "border": "0px",
              "height": "35px",
              "width": "45px",
            }
          }
          type="button"
        >
          <MdAddShoppingCart
            size={20}
            style={
              Object {
                "color": "#ffffff",
              }
            }
          >
            <IconBase
              size={20}
              style={
                Object {
                  "color": "#ffffff",
                }
              }
              viewBox="0 0 40 40"
            >
              <svg
                fill="currentColor"
                height={20}
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "color": "#ffffff",
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 40 40"
                width={20}
              >
                <g>
                  <path
                    d="m12 24.6q0 0.4 0.3 0.4h19.3v3.4h-20q-1.3 0-2.3-1.1t-0.9-2.3q0-0.8 0.4-1.6l2.2-4.1-6-12.7h-3.4v-3.2h5.5q1.6 3.2 3.1 6.6 0.4 0.7 1.8 3.7t2.2 4.7h11.7q5.9-10.7 6.4-11.8l2.9 1.7-6.4 11.6q-0.9 1.7-2.9 1.7h-12.4l-1.5 2.8z m16.4 5.4q1.3 0 2.3 1t0.9 2.4-0.9 2.3-2.3 0.9-2.4-0.9-1-2.3 1-2.4 2.4-1z m-16.8 0q1.4 0 2.4 1t1 2.4-1 2.3-2.4 0.9-2.3-0.9-0.9-2.3 0.9-2.4 2.3-1z m6.8-15v-5h-5v-3.4h5v-5h3.2v5h5v3.4h-5v5h-3.2z"
                  />
                </g>
              </svg>
            </IconBase>
          </MdAddShoppingCart>
        </button>
      </div>
    </div>
  </div>
</card>
`;
